User profile upgrade

kaysoky opened the pull request:
- UserProfile stores a epoch timestamp for Terms Of Use signing time. 
- UserProfile no longer requires ObjectSchema to use

kaysoky committed:
Change UserGroupDAO's getForUpdate to getEtagForUpdate

kaysoky committed:
PLFM-1756

- Changed UserProfile serialization from Schema-based to JDO-based
- Changed UserProfileManager to not require schemas

kaysoky committed:
Term of Use

Last updated sometime around March 2013.  Timestamp is placed a bit before just in case.

kaysoky committed:
UserProfile code review changes

- Terms of Use re-signing is no longer forced, but the option is there
- Old UserProfile serialization method is fully deprecated (moved into test)

kaysoky committed:
Added boolean for AgreesToTermsOfUse to UserProfile

brucehoff commented on services/workers/src/main/java/org/sagebionetworks/crowd/workers/CrowdGroupSynchronizer.java:
```diff
@@ -72,32 +71,48 @@ public void run() {
+					UserProfile userProfile = userProfileDAO.get(principalId);
+					
+					// For profiles that already exist, migrate the boolean for the termsOfUse over
+					if (userProfile.getAgreesToTermsOfUse() == null) {
```
missing case: Crowd boolean changes between worker runs.

kaysoky committed:
Smaller code review

- Added missing case for Crowd -> RDS migration of UserProfile
- Removed commented-out line in utils

kaysoky committed:
Changes to exception handling
